### PR TITLE
SLURM cluster on SLES 12 HPC SKU

### PR DESCRIPTION
**Pull request by smith1511**
Creates a fully configured SLURM cluster based on the SLES 12 HPC SKU.  This cluster is ready to run Intel MPI workloads when deployed on A8/A9 instances.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@smith1511__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by azurecla** [comment]
__@smith1511__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, AZPRBOT;

File: slurm-on-sles12-hpc/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,559 @@
...
+    "subnetRef": "[concat(variables('vnetID'),'/subnets/',variables('networkSettings').subnet.dse.name)]",
+    "templateBaseUrl": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/slurm-on-sles12-hpc/",
+    "installationCLI": "[concat('bash azuredeploy.sh ', variables('masterVMName'), ' ', variables('workerVMName'), ' ', parameters('workerNodeCount'), ' ', parameters('hpcUserName'), ' ', variables('templateBaseUrl'))]",
+    "storageAccountType": "Standard_LRS"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[parameters('newStorageAccountName')]",
+      "apiVersion": "2015-05-01-preview",
```
Comment (singhkay) [inline slurm-on-sles12-hpc/azuredeploy.json]: I suggest making the api version a variable and using it for all resources here and below. Also use the GA api 2015-06-15

File: slurm-on-sles12-hpc/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,559 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "dnsName": {
```
Comment (singhkay) [inline slurm-on-sles12-hpc/azuredeploy.json]: dnsname and storageaccounts cause a bunch of template failures due to strict requirements, I suggest creating variables for these and generating these using the uniqueString() function. See usage here https://github.com/Azure/azure-quickstart-templates/blob/bfb0c40965969a4884ccee65ff75c7ee13e3195b/301-2fe-lb80-rdp-1be-nsg-rdp/azuredeploy.json#L32

**Comment by smith1511** [comment]
Comments addressed...

**Comment by singhkay** [comment]
Thanks for the update @smith1511. Can you check the comment on using uniquestring for dns name as well?

**Comment by smith1511** [comment]
Can you point me at another example using uniqueString for DNS name?  

I'm unsure about this change to be honest.  I can understand storage as it's essentially an internal impl detail.  However the DNS name is the 'public face' to the customers service/RG and they may in fact want to choose it.  (Unless the example allows it to be overridden?)

**Comment by singhkay** [comment]
@smith1511 Good point, I don't have an example atm but this is something we're looking at solving since we're seeing so many folks running into failures that look like below where they entered the fqdn instead of a prefix. Maybe renaming the parameter to say dnsPrefix would be a good solution for now?

The domain name label test.somedomain.rd.us is invalid. It must conform to the following regular expression: ^[a-z][a-z0-9-]{1,61}[a-z0-9]$

**Comment by smith1511** [comment]
done

**Pull request merged**